src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                    methodSymbol.Parameters.Length == 0 && methodSymbol.ReturnType?.SpecialType == SpecialType.System_Int32)
+                {
+                    // before var count = (from a in b select a).Count();
+                    // after
+                    // var count = 0;
+                    // foreach (var a in b)
+                    // {
+                    //     count++;
+                    //  }
+                    return TryConvertIfInInvocation(
+                            source,
+                            invocationExpression,
+                            (variableIdentifier, expression) => SyntaxFactory.ExpressionStatement(
+                                SyntaxFactory.PostfixUnaryExpression(SyntaxKind.PostIncrementExpression, variableIdentifier)), // Generating 'count++'
+                            SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal(0)), // count = 0
+                            variableName: "count",
+                            out documentUpdateInfo);
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInToListInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // before var list = (from a in b select a).ToList();
+                // after
+                // var list = new List<T>();
+                // foreach (var a in b)
+                // {
+                //     list.Add(a)
+                // }
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0)
+                {
+                    return TryConvertIfInInvocation(
+                              source,
+                              invocationExpression,
+                              (listIdentifier, expression) => SyntaxFactory.ExpressionStatement(SyntaxFactory.InvocationExpression(
+                                    SyntaxFactory.MemberAccessExpression(
+                                        SyntaxKind.SimpleMemberAccessExpression,
+                                        listIdentifier,
+                                        SyntaxFactory.IdentifierName(nameof(IList.Add))),
+                                    SyntaxFactory.ArgumentList(SyntaxFactory.SingletonSeparatedList(SyntaxFactory.Argument(expression))))),
+                              SyntaxFactory.ObjectCreationExpression(
+                                  methodSymbol.GenerateReturnTypeSyntax().WithAdditionalAnnotations(Simplifier.Annotation),
+                                  SyntaxFactory.ArgumentList(),
+                                  initializer: null),
+                               variableName: "list",
```

Comment:
>list [](start = 46, length = 4)

Are we sure the name is not going to conflict with anything in scope (local, parameter, field, property, method, ... ) ?

Reply:
yes, we check it with GetFreeSymbolName. We have tests for this. "list" is just a prefix.

---
In reply to: [176929183](https://github.com/dotnet/roslyn/pull/25362#discussion_r176929183) [](ancestors = 176929183)

Reply:
> yes, we check it with GetFreeSymbolName. We have tests for this. "list" is just a prefix.

Do we have any test where ```list``` is used in a nested scope?

---
In reply to: [177553404](https://github.com/dotnet/roslyn/pull/25362#discussion_r177553404) [](ancestors = 177553404,176929183)